### DISCUSSION TITLE
About plugin download request address

### DISCUSSION THREAD

**Discussion by guoxinghai** [sbt/sbt#7649] [Q&A] [answered]
I used **resolvers += Resolver.url()** to configure a mirror address in build.sbt and plugins.sbt.
But I found that some of the plugins are still downloaded through "https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/".
Here are the plugins downloaded from the above address.

1. sbt-compat.jar
2. sbt-pgp.jar
3. sbt-buildinfo.jar
4. sbt-dynver.jar
5. sbt-unidoc.jar
6. sbt-assembly.jar
7. pgp-library_2.12.jar

I'm sure these plugins are included in the mirror.

**Comment by eed3si9n** [comment, marked as answer]
If you want to take over the network call, likely you want to follow https://www.scala-sbt.org/1.x/docs/Proxy-Repositories.html.

**Reply by guoxinghai** [reply]
My guess is that the answer might have something to do with that.

![image](https://github.com/user-attachments/assets/2cbc02b1-c199-4887-8051-9dad5b198271)
sbt.repository.config as follow:

[repositories]
  local
  maven-local
  public.repo: https://maven.test.cn/repository/maven-public/
  public.repo-ivy: https://maven.test.cn/repository/maven-public/, [organization]/[module]/(scala_[scalaVersion]/)(sbt_[sbtVersion]/)[revision]/[type]s/[artifact](-[classifier]).[ext]


public.repo and public.repo-ivy point to the same address.

But I should point out that the link in this repository is not a proxy address, but a maven repository.

**Reply by guoxinghai** [reply]
One plugin (sbt-dependency-graph-0.9.2.jar) uses the configured address download, and the rest(sbt-compat.jar, sbt-pgp.jar ....) uses the remote address ：https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/

**Reply by eed3si9n** [reply]
sbt-dependency-graph has been in-sourced to sbt itself.

**Reply by guoxinghai** [reply]
> sbt-dependency-graph has been in-sourced to sbt itself.

Thank you for your answer. Is there any way to download other plug-ins through the address I configured (the required plug-ins have been stored in the address)?
